### DISCUSSION TITLE
FullWindowOverlay and native iOS Toast from burnt

### DISCUSSION THREAD

**Discussion by wcastand** [software-mansion/react-native-screens#1907] [Q&A] [answered]
trying to use Burnt but the native toasts are hidden behind my `FullWindowOverlay` component. i need it because it's a pin code security so need to go above all other window, modal included but not toasts x)

asked on Burnt discord in case they know what could be done but also asking here just in case if anyone faced that issue and found a solution.

**Comment by tboba** [comment]
Hi @wcastand! Can you tell me step by step what is your expected scenario? At first glance this looks like a bug, but I need to see full picture of your situation 😄 Also, are you trying to use Android or iOS?

**Reply by wcastand** [reply]
`FullWindowOverlay` is only used on iOS. 

i have a screen inside a `FullWindowOverlay` that is used when the user comeback after X seconde to ask for a security pin code, since i need this screen as a security it's over everything in the app and i used `FullWindowOverlay` to achieve that.

when we tried to use [burnt](https://github.com/nandorojo/burnt), the toast it creates is also hidden by the `FullWindowOverlay` screen, even if i call the toast inside the `FullWindowOverlay` component.

i'm no expert but from what i understood, burnt use the app root view as the parent for the toast but `FullWindowOverlay` create a view above the root to be above everything which means it's also above the toast.

**Reply by tboba** [reply]
> FullWindowOverlay is only used on iOS.

Good, I just wanted to ensure that Android is not the case here 😁 

> I'm no expert but from what i understood, burnt use the app root view as the parent for the toast but FullWindowOverlay create a view above the root to be above everything which means it's also above the toast.
That's right - when FullWindowOverlay is being mounted it's placed not inside `UITransitionView` (which is a view that holds the whole application UI) but in a separate container.

If it's possible for you to open your app in Xcode, could you open `View Hierarchy` and attach an image how your hierarchy looks like in debug navigator? You can hide your own views from it, just wanna make sure where the toast really is 😄

**Comment by wcastand** [comment]
guessing, RNSFullWindowOverlayContainer is `FullWindowOverlay` and SPIndicatorView is the toast

![Screenshot 2023-10-18 at 12 52 43](https://github.com/software-mansion/react-native-screens/assets/2178244/2c4990bd-2bd4-462e-aaac-83cb742d9037)

**Reply by wcastand** [reply]
![Screenshot 2023-10-18 at 12 56 19](https://github.com/software-mansion/react-native-screens/assets/2178244/7de7b518-ab06-4728-8eed-3b96455b73af)

it's the view hierarchy when the toast is hidden behind the full window overlay (i called the toast inside the full window overlay component and it's hidden behind it still)

**Reply by tboba** [reply, marked as answer]
Oh, yeah now it's clear why the toasts are being hidden underneath the FullWindowOverlay 😅 As I qualify this as a bug, I'll create an issue from the discussion (#1930). Thanks for reporting that!
